docs/source/chaincode4ade.rst

```diff
@@ -39,18 +39,27 @@ In each language, the ``Invoke`` method is called by clients to submit transacti
 proposals. This method allows you to use the chaincode to read and write data on
 the channel ledger.
 
-You also need to include an ``Init`` method that will serve as the initialization
-function for your chaincode. This method will be called in order to initialize
-the chaincode when it is started or upgraded. By default, this function is never
-executed. However, you can use the chaincode definition to request that the ``Init``
-function be executed. If execution of ``Init`` is requested, fabric will ensure
-that ``Init`` is invoked before any other function and is only invoked once.
-This option provides you additional control over which users can initialize the
-chaincode and the ability to add initial data to the ledger. If you are using
-the peer CLI to approve the chaincode definition, use the ``--init-required``
-flag to request the execution of the ``Init`` function. Then call the ``Init``
-function by using the `peer chaincode invoke` command and passing the
-``--isInit`` flag. For more information, see :doc:`chaincode_lifecycle`.
+If you use the low level APIs provided by the Fabric Chaincode Shim API, you need
+to include an ``Init`` method in your chaincode that will serve as the initialization
+function. This function is required by the chaincode interface, but does not
+need to invoked by your applications. The chaincode will start and be ready to use
+after the chanced definition is committed to the channel, without the ``Init`` function
+being called. However, you can request that ``Init`` become an executable function.
+Fabric will ensure that the ``Init`` function is invoked before any other function
+in the chaincode and is only invoked once. Requesting the execution of the ``Init``
+function allows you to implement logic that is run when the chaincode is initialized.
+You will need to call ``Init`` to initialize the chaincode every time the
+chaincode is started or upgraded.
+
+You can use the chaincode definition to request the execution of the ``Init``
+function by using the ``--init-required`` flag. You can then call the ``Init``
+function by using the ``peer chaincode invoke`` command and passing the ``--isInit``
+flag. For more information, see :doc:`chaincode_lifecycle`.
+
+If you are using the Fabric contract API, you do not need to include an ``Init``
+method in your chaincode. If you use the ``--init-required`` flag to request the
+that the chaincode be initialized by a call from your applications, you can
+initialize the chaincode by calling any function in the chaincode.
```

Comment:
```suggestion
If you use the low level APIs provided by the Fabric Chaincode Shim API, you need
to include an ``Init`` method in your chaincode that will serve as the initialization
function. This function is required by the chaincode interface, but does not
necessarily need to invoked by your applications. When you define your chaincode on a channel,
you can specify whether ``Init`` must be called prior to Invokes. If you specify that ``Init`` is required,
Fabric will ensure that the ``Init`` function is invoked before any other function
in the chaincode and is only invoked once. Requesting the execution of the ``Init``
function allows you to implement logic that is run when the chaincode is initialized,
for example to set some initial state.
You will need to call ``Init`` to initialize the chaincode every time you increment the version
of a chaincode, assuming the chaincode definition that increments the version indicates that ``Init`` is required.

If using the Fabric peer CLI, when approving and committing the chaincode definition, you can use the
``--init-required`` flag to indicate that ``Init`` must be called for the new chaincode version.
To call ``Init`` using the Fabric peer CLI, use the ``peer chaincode invoke`` command and pass the ``--isInit``
flag. For more information, see :doc:`chaincode_lifecycle`.

If you are using the Fabric contract API, you do not need to include an ``Init``
method in your chaincode. If you use the ``--init-required`` flag to request the
that the chaincode be initialized by a call from your applications, you can
initialize the chaincode by calling any function in the chaincode.
```

Reply:
I'm suggesting some edits to the first two paragraphs.

Concerning the 3rd paragraph, I actually don't understand how the contract API works in this regard. Does the contract API somehow indicate which of the functions serves the Init purpose?

Reply:
Good edits. I added them to the paragraphs in question. I also provide more detail about the Fabric contract API and the ``init-required`` behavior.